de-lib/src/main/java/org/iplantc/de/fileViewers/client/callbacks/GenomeBrowserUtil.java

```diff
@@ -0,0 +1,175 @@
+import com.google.common.collect.Lists;
+import com.google.gwt.safehtml.shared.SafeHtmlUtils;
+import com.google.gwt.user.client.TakesValue;
+import com.google.gwt.user.client.rpc.AsyncCallback;
+import com.google.gwt.user.client.ui.HTML;
+import com.google.inject.Inject;
+import com.google.web.bindery.autobean.shared.AutoBeanCodex;
+
+import com.sencha.gxt.core.shared.FastMap;
+import com.sencha.gxt.widget.core.client.container.VerticalLayoutContainer;
+import com.sencha.gxt.widget.core.client.event.HideEvent;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class GenomeBrowserUtil {
+
+    public interface GenomeBrowserUtilAppearance {
+
+        String indexFileMissing();
+
+        String indexFileMissingError();
+
+        String selectIndexFile();
+    }
+
+    @Inject
+    GenomeBrowserUtilAppearance appearance;
+    @Inject
+    DiskResourceErrorAutoBeanFactory factory;
+    @Inject
+    DiskResourceUtil diskResourceUtil;
+    @Inject
+    AsyncProviderWrapper<FileSelectDialog> dialogAsyncProviderWrapper;
+    @Inject
+    DiskResourceServiceFacade diskResourceServiceFacade;
+
+    @Inject
+    public GenomeBrowserUtil() {
+
+    }
+
+    public void sendToGenomeBrowser(List<DiskResource> resourcesToSend, IsMaskable container) {
+        final FastMap<TYPE> pathMap = new FastMap<>();
+
+        for (DiskResource resource : resourcesToSend) {
+            final String path = resource.getPath();
+            final String infoType = resource.getInfoType();
+
+            pathMap.put(path, TYPE.FILE);
+
+            String indexFilePath = null;
+            if (InfoType.BAM.toString().equals(infoType)) {
+                indexFilePath = path + ".bai";
+            } else if (InfoType.VCF.toString().equals(infoType)) {
+                indexFilePath = path + ".idx";
+            }
+
+            if (!Strings.isNullOrEmpty(indexFilePath)) {
+                pathMap.put(indexFilePath, TYPE.FILE);
+            }
+        }
+
+        diskResourceServiceFacade.getStat(pathMap, new DataCallback<FastMap<DiskResource>>() {
+            @Override
+            public void onFailure(Integer statusCode, Throwable caught) {
+                processError(caught, resourcesToSend, container);
+                if (container != null) {
+                    container.unmask();
+                }
+            }
+
+            @Override
+            public void onSuccess(FastMap<DiskResource> result) {
+                shareWithAnon(Lists.newArrayList(pathMap.keySet()), container);
+            }
+        });
+    }
+
+    protected void processError(Throwable caught,
+                                List<DiskResource> resourcesToSend,
+                                IsMaskable container) {
+        ErrorGetStat egs = AutoBeanCodex.decode(factory, ErrorGetStat.class, caught.getMessage()).as();
+        IPlantDialog dialog = new IPlantDialog();
+        dialog.setHeading(SafeHtmlUtils.fromTrustedString(appearance.indexFileMissing()));
```

Comment:
Not necessary for this PR, but ideally it'd be nice to refactor this dialog out into its own class so we can use `AsyncProviderWrapper`, add some static IDs, etc.

Reply:
👍